packages/server/test/adapters/aws-lambda.test.tsx

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { z } from 'zod';
+import * as trpc from '../../src';
+import { inferAsyncReturnType } from '../../src';
+import * as trpcLambda from '../../src/adapters/aws-lambda';
+import { mockAPIGatewayProxyEvent } from './aws-lambda.utils';
+
+const createContext = async ({
+  event,
+}: trpcLambda.CreateLambdaContextOptions) => {
+  return {
+    user: event.headers['X-USER'],
+  };
+};
+
+type Context = inferAsyncReturnType<typeof createContext>;
+const router = trpc
+  .router<Context>()
+  .query('hello', {
+    input: z
+      .object({
+        who: z.string().nullish(),
+      })
+      .nullish(),
+    resolve({ input, ctx }) {
+      return {
+        text: `hello ${input?.who ?? ctx.user ?? 'world'}`,
+      };
+    },
+  })
+  .query('echo', {
+    input: z.object({
+      who: z.object({ name: z.string().nullish() }),
+    }),
+    resolve({ input }) {
+      return {
+        text: `hello ${input.who.name}`,
+      };
+    },
+  });
+
+const handler = trpcLambda.createApiGatewayHandler({
+  router,
+  createContext,
+});
+
+test('basic test', async () => {
+  const result = await handler(
+    mockAPIGatewayProxyEvent({
+      body: JSON.stringify({}),
+      headers: { 'Content-Type': 'application/json', 'X-USER': 'Lilja' },
+      method: 'GET',
+      path: 'hello',
+    }),
+    {},
+  );
+  const body = JSON.parse(result.body);
+  delete (result as any).body;
+  expect(result).toMatchInlineSnapshot(`
+Object {
+  "headers": Object {
+    "Content-Type": "application/json",
+  },
+  "statusCode": 200,
+}
+`);
+  expect(body).toMatchInlineSnapshot(`
```

Comment:
Can avoid the `any`/usage of `delete` here:

```suggestion
  const { body, ...result } = await handler(
    mockAPIGatewayProxyEvent({
      body: JSON.stringify({}),
      headers: { 'Content-Type': 'application/json', 'X-USER': 'Lilja' },
      method: 'GET',
      path: 'hello',
    }),
    {},
  );
  expect(result).toMatchInlineSnapshot(`
Object {
  "headers": Object {
    "Content-Type": "application/json",
  },
  "statusCode": 200,
}
`);
  expect(JSON.parse(body)).toMatchInlineSnapshot(`
```

Same for the others in this file.